From 109000393cf142ff9f596b1b8166b51a34d9a849 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 10 Jun 2015 09:30:02 +0530
Subject: [PATCH] [EM]: Fix for KW issue #88493 and #88494

The struct fusb300_int_stat's variable int_stat is uninitialized. To
fix the kw issue, used memset to initialize the int_stat after the
variable declaration.

Change-Id: Ic87df93412ce1039a00fd2286ab2dd148cf26b03
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38983
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 83a2caae8cd2..2276bd398d26 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -741,6 +741,7 @@ static irqreturn_t fusb300_interrupt(int id, void *dev)
 	unsigned int reg_start;
 	size_t count;
 
+	memset(&int_stat, 0x00, sizeof(int_stat));
 	pm_runtime_get_sync(chip->dev);
 
 	if (chip->is_fusb300) {
-- 
1.9.1

